lib/minicbor-lease/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! An adapter implementing [`minicbor::encode::write::Write`] for
+//! [`idol_runtime::Leased`] byte buffers.
+
+#![no_std]
+
+/// An adapter implementing [`minicbor::encode::write::Write`] for
+/// [`idol_runtime::Leased`] byte buffers.
+pub struct LeasedWriter<'lease, A>
+where
+    A: idol_runtime::AttributeWrite,
+{
+    lease: &'lease mut idol_runtime::Leased<A, [u8]>,
+    pos: usize,
+    ran_out_of_space: bool,
```

Comment:
Is this still needed, now that we can get the inner error?  I see it written, but no one ever actually checks it.